qemu/tests/qemu_guest_agent.py

```diff
@@ -4168,15 +4185,52 @@ def _base_on_bg_check_resource_leak():
 
             if bg.is_alive():
                 qga_handles, qga_memory = _process_qga_resource()
-                # Generally the qga handles won't exceed 300, and the memory
-                # setting '3000' is an average value when the guest is not
-                # running redundant operations and apps.
-                if qga_handles > 300 or qga_memory > 3000:
-                    test.fail("QGA commands caused resource leak. handles are"
-                              " %s, memory is %s" % (qga_handles, qga_memory))
-                else:
-                    LOG_JOB.info("Current qga handles is %s, qga memory"
-                                 "is %s", qga_handles, qga_memory)
+                if qga_memory > qga_mem_threshold:
+                    qga_memory_previous = qga_memory
+                    qga_handles_previous = qga_handles
+                    qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                    memory_high_peak = []
+                    memory_low_peak = []
+                    memory_list = []
+                    start_time = time.time()
+                    while int(time.time() - start_time) < qga_mem_leak_timeout:
+                        while qga_memory_latest >= qga_memory_previous:
+                            difference = qga_memory_latest - qga_memory_previous
+                            qga_memory_previous = qga_memory_latest
+                            time.sleep(5)
+                            qga_handles_latest, qga_memory_latest = _process_qga_resource()
+                            memory_list.append(qga_memory_latest)
+                            if qga_memory_latest < qga_memory_previous:
```

Comment:
```suggestion
                            if qga_memory_latest < qga_memory_previous or stable_count > 5:
                                stable_count = 0
```
Probably not 5 but even smaller.